code/modules/SCP/SCPs/SCP-294.dm

```diff
@@ -1,104 +1,137 @@
-	icon = 'icons/obj/scp294.dmi'
+	icon = 'icons/SCP/scp294.dmi'
+
 	icon_state = "coffee_294"
-	layer = 2.9
 	anchored = TRUE
 	density = TRUE
-	SCP = /datum/scp/scp_294
-
-	var/usage_cooldown
-	var/usage_cooldown_time = 40 SECONDS
-
-	/// These reagents cannot be dispensed under any circumstances
-	var/list/banned_chems = list(
-	/datum/reagent/adminordrazine,
-	/datum/reagent/nitroglycerin,
-	/datum/reagent/glycerol,
-	/datum/reagent/coolant,
-	/datum/reagent/frostoil,
-	/datum/reagent/zombie,
-	/datum/reagent/jerraman,
-	/datum/reagent/three_eye,
-	/datum/reagent/mutation_toxin,
-	/datum/reagent/advanced_mutation_toxin,
-	/datum/reagent/grauel,
-	/datum/reagent/laich,
-	/datum/reagent/gottheit,
-	)
-	/// List of name shortcuts for allowed chems: Name = Datum
+
+	//Config
+
+	///How long to wait until we can use again
+	var/usage_cooldown = 15 MINUTES
+	///How many times we can use the machine before it needs to cooldown
+	var/max_uses = 30
+
+	//Mechanical
+
+	///Tracks our usage cooldown
+	var/usage_cooldown_tracker
+	///Tracks our uses
+	var/uses_tracker
+	///Have we started restock?
+	var/started_restock = FALSE
+	///Reagent types
+
+	///Shortcut names for things like "poision" or "death" where it dosent match a reagent name exactly
 	var/list/shortcut_chems = list(
-		// Medicine
-		"inap" = /datum/reagent/medicine/inaprovaline,
-		"tric" = /datum/reagent/medicine/tricordrazine,
-		"bica" = /datum/reagent/medicine/bicaridine,
-		"mera" = /datum/reagent/medicine/meraline,
-		"kelo" = /datum/reagent/medicine/kelotane,
-		"derma" = /datum/reagent/medicine/dermaline,
-		"dexa" = /datum/reagent/medicine/dexalin,
-		"dexa+" = /datum/reagent/medicine/dexalin_plus,
-		"dylo" = /datum/reagent/medicine/dylovene,
-		"vena" = /datum/reagent/medicine/dylovene/venaxilin,
-		"hyro" = /datum/reagent/medicine/hyronalin,
-		"peri" = /datum/reagent/medicine/peridaxon,
-		"alky" = /datum/reagent/medicine/alkysine,
-		// Drugs/Narcotics
-		"crypto" = /datum/reagent/cryptobiolin,
-		"imped" = /datum/reagent/impedrezene,
-		// Poison
-		"venom" = /datum/reagent/toxin/venom,
 		"spider" = /datum/reagent/toxin/venom,
-		// Sedatives
 		"sleep" = /datum/reagent/soporific,
-		"chloral" = /datum/reagent/chloral_hydrate,
-		"zombie" = /datum/reagent/toxin/zombie_powder,
+		"kidnap" = /datum/reagent/chloral_hydrate,
+		"zombie" = /datum/reagent/scp008,
+		"death" = /datum/reagent/toxin/cyanide,
+		"joe" = /datum/reagent/blood,
+		"008" = /datum/reagent/scp008,
+		"health" = /datum/reagent/scp500
 	)
+/obj/machinery/scp294/Initialize()
+	. = ..()
+	SCP = new /datum/scp(
+		src, // Ref to actual SCP atom
+		"coffee machine", //Name (Should not be the scp desg, more like what it can be described as to viewers)
+		SCP_EUCLID, //Obj Class
+		"294", //Numerical Designation
+	)
+//Mechanics
+
+///Cycles through all reagents datums and picks out ones that contain the chemical we are looking for
```

Comment:
This may break a lot of things, leading to empty food/pills/medipens.
I am honestly not sold on this idea, as you also removed the blacklisted chems, making taking SCP500 and 008 incredibly easy.

Reply:
294 will have to be moved to a more restricted area, like LCZ.

Also, I will have interactions with containers that 294 is transferring from have a visual effect + some sort of message.

Ill also probably have pills, injectors, and other one-use chem containers delete themselves if they become empty.

Reply:
I'll do this in a PR right after this one as I dont want delay this being merged any further